src/components/FileTreeNode/index.tsx

```diff
@@ -73,4 +179,19 @@ class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
   }
 }
 
-export default FileTreeNodeBase;
+const mapStateToProps = (state: ApplicationState, ownProps: PublicProps) => {
+  const { versionId } = ownProps;
+
+  const version = getVersionInfo(state.versions, versionId);
+  const selectedPath = version ? version.selectedPath : undefined;
+  const linterMessages = version
```

Comment:
This patch introduces the same problem as https://github.com/mozilla/addons-code-manager/pull/400 but for `linterMessages`. The problem is that the component maps data from Redux but it doesn't know how to load that data when it doesn't exist. 

If this component can't load its own data then it has an implicit dependency on loader components such as [`Browse`](https://github.com/mozilla/addons-code-manager/blob/cd51cf5cea56138961e9eec7ff7d9ac193320369/src/pages/Browse/index.tsx#L86) or [`DiffView`](https://github.com/mozilla/addons-code-manager/blob/cd51cf5cea56138961e9eec7ff7d9ac193320369/src/components/DiffView/index.tsx#L104). The dependency is not explicit which makes it a fragile relationship. Let's say I move the loading logic *out* of `Browse` (which I plan to do); I would probably forget to check that this component still works.

In https://github.com/mozilla/addons-code-manager/pull/400 I suggested passing a direct public prop for what you need but if you did that here, you'd have to prop-drill `linterMessages` through `FileTree` which is not great. 

The best solution would be to implement [`<LinterProvider>`](https://github.com/mozilla/addons-code-manager/issues/414) because that's exactly what you need here. 

<details>
<summary>example patch</summary>

```diff
diff --git a/src/components/FileTreeNode/index.tsx b/src/components/FileTreeNode/index.tsx
index 092f273..dd976fc 100644
--- a/src/components/FileTreeNode/index.tsx
+++ b/src/components/FileTreeNode/index.tsx
@@ -92,14 +92,13 @@ const getIconForType = (type: string | null) => {
 };
 
 export class FileTreeNodeBase<TreeNodeType> extends React.Component<Props> {
-  render() {
+  renderWithMessages(linterMessages) {
     const {
       getToggleProps,
       hasChildNodes,
       isExpanded,
       isFolder,
       level,
-      linterMessages,
       node,
       onSelect,
       renderChildNodes,
@@ -177,6 +176,16 @@ export class FileTreeNodeBase<TreeNodeType> extends React.Component<Props> {
       </React.Fragment>
     );
   }
+
+  render() {
+    const { version } = this.props;
+
+    return (
+      <LinterProvider version={version}>
+        {(linterMessages) => this.renderWithMessages(linterMessages)}
+      </LinterProvider>
+    );
+  }
 }
 
 const mapStateToProps = (state: ApplicationState, ownProps: PublicProps) => {
```

</details>

It would introduce two linter message loaders on the same page but that's fine -- the first one to begin rendering will be the one that loads the data. The others will block on the `isLoading` flag.

The quickest solution would be to just use prop drilling and make a TODO comment to remove it once we have `<LinterProvider>`. I'm open to other ideas, though. Let me know if you have questions about why this is a problem.